fdbclient/ThreadSafeTransaction.cpp

```diff
@@ -482,15 +485,20 @@ ThreadFuture<Void> ThreadSafeTransaction::commit() {
 
 Version ThreadSafeTransaction::getCommittedVersion() {
 	// This should be thread safe when called legally, but it is fragile
```

Comment:
IIUC this is now thread safe when called illegally (i.e. before the commit future has succeeded). Should we update the comment?

Reply:
I was thinking about this too, and there may still be an issue here. In particular, the `committedVersion` value we are returning is just a plain `Version` that is being set from the network thread and being read from an arbitrary thread. I think what this is saying is that if you call this after committing, then there wouldn't be a race between the write and read, but if you call it during a commit then we may still not be safe.

Reply:
A fix for that could be to make the committed version atomic. We should do so, but I don't know if we want to put that in this PR or make it a separate one.

Reply:
Considering the [documentation](https://apple.github.io/foundationdb/api-c.html#c.fdb_transaction_get_committed_version) for `get_committed_version` states that the transaction must have been committed, and the biggest issue with calling `get_committed_version` before it is committed is that you might read garbage data, is it an issue to leave the current behavior? As far as I understand there should be no way to cause a crash when reading/writing from different threads, the worst that could happen is a read could happen in the middle of a write and read some halfway data.

Reply:
A data race is undefined behavior, so there's not much guarantee about what would happen. In practice it may be just that you get a garbage result, but even in that case it's probably friendlier if we return a more interpretable value. That said, this is not as high a priority as the other changes in this PR since it probably only manifests with illegal usage of the API.

Reply:
You're right AJ this is still a data race. I'm going to mark this conversation resolved.